products/statement-generator/src/tests/buttons.test.tsx

```diff
@@ -0,0 +1,144 @@
+    const { getByText, getByTestId } = render(
+      <ButtonComponent buttonText="Back" hasBackArrow />
+    );
+
+    const button = getByText(/Back/i);
+    const arrow = getByTestId('back-arrow');
+
+    expect(button).toBeTruthy();
+    expect(arrow).toBeTruthy();
+  });
+
+  test('Button renders forward arrow', () => {
+    const { getByText, getByTestId } = render(
+      <ButtonComponent buttonText="Next" hasForwardArrow />
+    );
+
+    const button = getByText(/Next/i);
+    const arrow = getByTestId('forward-arrow');
+
+    expect(button).toBeTruthy();
+    expect(arrow).toBeTruthy();
+  });
+
+  // Test that the button is disabled when the disabled prop is true
+  test('Button is disabled when disabled prop is true', () => {
+    const { getByRole } = render(
+      <ButtonComponent buttonText="Disabled Button" disabled={true} />
+    );
+
+    const button = getByRole('button', { name: /Disabled Button/i });
+
+    if (button) {
+      expect(button).toBeDisabled();
+    }
+  });
+
+  // Test that the button is not disabled when the disabled prop is false
+  test('Button is re-enabled when disabled prop is false', () => {
+    const { getByRole } = render(
+      <ButtonComponent buttonText="Enabled Button" disabled={false} />
+    );
+
+    const button = getByRole('button', { name: /Enabled Button/i });
+
+    if (button) {
+      expect(button).not.toBeDisabled();
+    }
+  });
+
+  // Test that the button has appropriate styles when disabled
+  test('Disabled button has appropriate styles', () => {
+    const { getByRole } = render(
+      <ButtonComponent buttonText="Disabled Button" disabled={true} />
+    );
+
+    const button = getByRole('button', { name: /Disabled Button/i });
+
+    expect(button).toHaveStyle({ color: 'rgba(0, 0, 0, 0.26)' });
+    expect(button).toHaveStyle({ background: 'rgb(197, 179, 209)' });
+  });
+
+  // Test that the button has appropriate styles when not disabled
+  test('Non-disabled button has appropriate styles', () => {
+    const { getByRole } = render(
+      <ButtonComponent buttonText="Enabled Button" disabled={false} />
+    );
+
+    const button = getByRole('button', { name: /Enabled Button/i });
+
+    expect(button).not.toHaveStyle({ color: 'rgba(0, 0, 0, 0.26)' });
+    expect(button).not.toHaveStyle({ background: 'rgb(197, 179, 209)' });
+  });
+
+  // Test that the button is at least 44px wide
+  test('Button is at least 44px', () => {
+    const { getByRole } = render(<ButtonComponent buttonText="Click me" />);
+
+    const button = getByRole('button', { name: /Click me/i });
+
+    expect(button).toHaveStyle({ minHeight: '44px' });
+  });
+
+  // Test that the button is focusable
+  test('Button can receive focus', () => {
+    const { getByRole } = render(<ButtonComponent buttonText="Click me" />);
+
+    const button = getByRole('button', { name: /Click me/i });
+
+    fireEvent.focus(button);
+
+    expect(button).toHaveClass('Mui-focusVisible');
```

Comment:
Rather than checking for the related MUI class, are we able to check if the button itself is focused?

Reply:
@sydneywalcoff I had a lot of trouble with with the focusing at first, mostly because the mui Button component is technically a span and not a button. I did get it working in the end though, so I'll try and see if I can find an alternative to looking for this class!

Reply:
@sydneywalcoff My latest commit checks for focus more directly!